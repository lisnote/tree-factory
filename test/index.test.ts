import { expect, test } from 'vitest';
import { cloneDeep } from 'lodash-es';
import {
  treeCleaner,
  treeFind,
  treeFromList,
  treeToList,
  treeTraverse,
} from '../src/index.ts';

const source = [
  { id: 1, value: 1 },
  { id: 0, value: 0, pid: 3 },
  { id: 3, value: 3 },
  { id: 2, value: 2, pid: 1 },
  { id: 4, value: 4, pid: 1 },
];
const tree = treeFromList(source);
const cloneTree = () => cloneDeep(tree);

test('treeFromList', () => {
  expect(tree).toMatchSnapshot();
});

test('treeToList', () => {
  expect(treeToList(tree)).toMatchSnapshot();
});

test('treeCleaner', () => {
  expect(
    treeCleaner(cloneTree(), (node) => node.data.value > 4),
  ).toMatchSnapshot();
});

test('treeTraverse', () => {
  const traverseTree = cloneTree();
  treeTraverse(traverseTree, (node) => node.data.value++);
  expect(traverseTree).toMatchSnapshot();
});

test('treeFind', () => {
  expect(treeFind(tree, (node) => node.data.value > 3)).toMatchSnapshot();
});
