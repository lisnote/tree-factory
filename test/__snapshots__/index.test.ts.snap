// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`treeCleaner 1`] = `[]`;

exports[`treeFind 1`] = `
{
  "children": [],
  "data": {
    "id": 4,
    "pid": 1,
    "value": 4,
  },
  "parent": {
    "children": [
      {
        "children": [],
        "data": {
          "id": 2,
          "pid": 1,
          "value": 2,
        },
        "parent": [Circular],
      },
      [Circular],
    ],
    "data": {
      "id": 1,
      "value": 1,
    },
  },
}
`;

exports[`treeFromList 1`] = `
[
  {
    "children": [
      {
        "children": [],
        "data": {
          "id": 2,
          "pid": 1,
          "value": 2,
        },
        "parent": [Circular],
      },
      {
        "children": [],
        "data": {
          "id": 4,
          "pid": 1,
          "value": 4,
        },
        "parent": [Circular],
      },
    ],
    "data": {
      "id": 1,
      "value": 1,
    },
  },
  {
    "children": [
      {
        "children": [],
        "data": {
          "id": 0,
          "pid": 3,
          "value": 0,
        },
        "parent": [Circular],
      },
    ],
    "data": {
      "id": 3,
      "value": 3,
    },
  },
]
`;

exports[`treeToList 1`] = `
[
  {
    "children": [
      {
        "children": [],
        "data": {
          "id": 2,
          "pid": 1,
          "value": 2,
        },
        "parent": [Circular],
      },
      {
        "children": [],
        "data": {
          "id": 4,
          "pid": 1,
          "value": 4,
        },
        "parent": [Circular],
      },
    ],
    "data": {
      "id": 1,
      "value": 1,
    },
  },
  {
    "children": [],
    "data": {
      "id": 2,
      "pid": 1,
      "value": 2,
    },
    "parent": {
      "children": [
        [Circular],
        {
          "children": [],
          "data": {
            "id": 4,
            "pid": 1,
            "value": 4,
          },
          "parent": [Circular],
        },
      ],
      "data": {
        "id": 1,
        "value": 1,
      },
    },
  },
  {
    "children": [],
    "data": {
      "id": 4,
      "pid": 1,
      "value": 4,
    },
    "parent": {
      "children": [
        {
          "children": [],
          "data": {
            "id": 2,
            "pid": 1,
            "value": 2,
          },
          "parent": [Circular],
        },
        [Circular],
      ],
      "data": {
        "id": 1,
        "value": 1,
      },
    },
  },
  {
    "children": [
      {
        "children": [],
        "data": {
          "id": 0,
          "pid": 3,
          "value": 0,
        },
        "parent": [Circular],
      },
    ],
    "data": {
      "id": 3,
      "value": 3,
    },
  },
  {
    "children": [],
    "data": {
      "id": 0,
      "pid": 3,
      "value": 0,
    },
    "parent": {
      "children": [
        [Circular],
      ],
      "data": {
        "id": 3,
        "value": 3,
      },
    },
  },
]
`;

exports[`treeTraverse 1`] = `
[
  {
    "children": [
      {
        "children": [],
        "data": {
          "id": 2,
          "pid": 1,
          "value": 3,
        },
        "parent": [Circular],
      },
      {
        "children": [],
        "data": {
          "id": 4,
          "pid": 1,
          "value": 5,
        },
        "parent": [Circular],
      },
    ],
    "data": {
      "id": 1,
      "value": 2,
    },
  },
  {
    "children": [
      {
        "children": [],
        "data": {
          "id": 0,
          "pid": 3,
          "value": 1,
        },
        "parent": [Circular],
      },
    ],
    "data": {
      "id": 3,
      "value": 4,
    },
  },
]
`;
